documentation for module-level `# type: ignore` is unclear

My expectation of a module-level `# type: ignore` comment is that it will skip *typechecking* of everything in the module; i.e., everything would be given the type `Any`.

The description in the documentation such a comment "has the effect of ignoring the *entire* module.", is, I think, consistent with that impression, since what "ignoring" means earlier in the documentation has been basically skipping typechecking.

What it *actually* does, though, seems to be to cause mypy to quite literally ignore the module: it is just skipped entirely, with the result that you *can't import names defined in that module in another one*:

```
(.venv3) BenW ~ 02:10:35 $ cat foo/a.py
# type: ignore

class A(object):
    pass
cat(.venv3) BenW ~ 02:10:45 $ cat foo/b.py
from foo.a import A

def b(a):
    # type: (A) -> int
    return len(str(type(a)))

print(b(A()))
(.venv3) BenW ~ 02:10:47 $ mypy foo
foo/b.py:1: error: Module 'foo.a' has no attribute 'A'
Found 1 error in 1 file (checked 3 source files)
```

This makes the feature somewhat less useful IMO, but also, I think it could be clearer in the docs that that's what it does.

--- a/docs/source/common_issues.rst
+++ b/docs/source/common_issues.rst
@@ -187,7 +187,11 @@ Ignoring a whole file
 ---------------------
 
 A ``# type: ignore`` comment at the top of a module (before any statements,
-including imports or docstrings) has the effect of ignoring the *entire* module.
+including imports or docstrings) has the effect of ignoring the entire contents of the module.
+
+To only ignore errors, use a top-level ``# mypy: ignore-errors`` comment instead.
+To only ignore errors with a specific error code, use a top-level
+``# mypy: disable-error-code=...`` comment.
 
 .. code-block:: python
 

--- a/docs/source/common_issues.rst
+++ b/docs/source/common_issues.rst
@@ -192,6 +192,8 @@ including imports or docstrings) has the effect of ignoring the entire contents 
 To only ignore errors, use a top-level ``# mypy: ignore-errors`` comment instead.
 To only ignore errors with a specific error code, use a top-level
 ``# mypy: disable-error-code=...`` comment.
+To replace the contents of the module with ``Any``, use a per-module ``follow_imports = skip``.
+See :ref:`Following imports <follow-imports>` for details.
 
 .. code-block:: python
 

--- a/mypy/fastparse.py
+++ b/mypy/fastparse.py
@@ -485,7 +485,7 @@ class ASTConverter:
             if self.type_ignores[min(self.type_ignores)]:
                 self.fail(
                     (
-                        "type ignore with error code is not supported at module level; "
+                        "type ignore with error code is not supported for modules; "
                         "use `# mypy: disable-error-code=...`"
                     ),
                     line=min(self.type_ignores),

--- a/mypy/fastparse.py
+++ b/mypy/fastparse.py
@@ -482,6 +482,16 @@ class ASTConverter:
             and self.type_ignores
             and min(self.type_ignores) < self.get_lineno(stmts[0])
         ):
+            if self.type_ignores[min(self.type_ignores)]:
+                self.fail(
+                    (
+                        "type ignore with error code is not supported at module level; "
+                        "use `# mypy: disable-error-code=...`"
+                    ),
+                    line=min(self.type_ignores),
+                    column=0,
+                    blocker=False,
+                )
             self.errors.used_ignored_lines[self.errors.file][min(self.type_ignores)].append(
                 codes.FILE.code
             )

--- a/test-data/unit/check-errorcodes.test
+++ b/test-data/unit/check-errorcodes.test
@@ -919,5 +919,5 @@ def f(d: D, s: str) -> None:
 
 
 [case testRecommendErrorCode]
-# type: ignore[whatever]  # E: type ignore with error code is not supported at module level; use `# mypy: disable-error-code=...`  [syntax]
+# type: ignore[whatever]  # E: type ignore with error code is not supported for modules; use `# mypy: disable-error-code=...`  [syntax]
 1 + "asdf"

--- a/test-data/unit/check-errorcodes.test
+++ b/test-data/unit/check-errorcodes.test
@@ -916,3 +916,8 @@ def f(d: D, s: str) -> None:
     d[s]  # type: ignore[literal-required]
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-typeddict.pyi]
+
+
+[case testRecommendErrorCode]
+# type: ignore[whatever]  # E: type ignore with error code is not supported at module level; use `# mypy: disable-error-code=...`  [syntax]
+1 + "asdf"

